pkg/templates/untrustedimages/template.go

```diff
@@ -0,0 +1,55 @@
+package untrustedimages
+
+import (
+	"fmt"
+	"regexp"
+
+	"golang.stackrox.io/kube-linter/pkg/check"
+	"golang.stackrox.io/kube-linter/pkg/config"
+	"golang.stackrox.io/kube-linter/pkg/diagnostic"
+	"golang.stackrox.io/kube-linter/pkg/objectkinds"
+	"golang.stackrox.io/kube-linter/pkg/templates"
+	"golang.stackrox.io/kube-linter/pkg/templates/untrustedimages/internal/params"
+	"golang.stackrox.io/kube-linter/pkg/templates/util"
+	v1 "k8s.io/api/core/v1"
+)
+
+const (
+	templateKey = "untrusted-images"
+)
+
+func init() {
+	templates.Register(check.Template{
+		HumanName:   "Untrusted images",
+		Key:         templateKey,
+		Description: "Flag applications running containers with untrusted container images",
+		SupportedObjectKinds: config.ObjectKindsDesc{
+			ObjectKinds: []string{objectkinds.DeploymentLike},
+		},
+		Parameters:             params.ParamDescs,
+		ParseAndValidateParams: params.ParseAndValidate,
+		Instantiate: params.WrapInstantiateFunc(func(p params.Params) (check.Func, error) {
+			return util.PerContainerCheck(func(container *v1.Container) []diagnostic.Diagnostic {
+
+				// checking whether the container image is a trusted or not
+
+				allowedResult, blockedResult := false, true
+
+				for _, allowed := range p.AllowedList {
+					allowedRe := regexp.MustCompile(allowed)
+					allowedResult = allowedRe.MatchString(container.Image)
+				}
+
+				for _, blocked := range p.BlockedList {
+					blockedRe := regexp.MustCompile(blocked)
+					blockedResult = blockedRe.MatchString(container.Image)
+				}
+
+				if blockedResult || !allowedResult {
+					return []diagnostic.Diagnostic{{Message: fmt.Sprintf("The container %q is using an untrusted image, %q.", container.Name, container.Image)}}
```

Comment:
We should make this message more useful.
For example:
1. If the image is explicitly blocked, then we should say which blocklist regex it matched.
2. If we're failing because it's not explicitly allowed, then we should say that.

Reply:
Noted. Thank you!